java-connectors/kafka-connect-azure-servicebus/src/test/java/io/lenses/streamreactor/connect/azure/servicebus/source/TaskToReceiverBridgeTest.java

```diff
@@ -0,0 +1,130 @@
+import org.junit.jupiter.api.Test;
+
+class TaskToReceiverBridgeTest {
+
+  Map<String, String> properties = new HashMap<>();
+  ArrayBlockingQueue<ServiceBusMessageHolder> blockingQueue;
+  OffsetStorageReader offsetReader;
+
+  TaskToReceiverBridge testObj;
+
+  @BeforeEach
+  void setUp() {
+    blockingQueue = mock(ArrayBlockingQueue.class);
+    offsetReader = mock(OffsetStorageReader.class);
+  }
+
+  @Test
+  void closeReceiversShouldCloseAllReceivers() {
+    //given
+    ExecutorService executorService = mock(ExecutorService.class);
+    String receiverId1 = "RECEIVER1";
+    String receiverId2 = "RECEIVER2";
+    ServiceBusReceiverFacade receiver1 = mock(ServiceBusReceiverFacade.class);
+    ServiceBusReceiverFacade receiver2 = mock(ServiceBusReceiverFacade.class);
+    when(receiver1.getReceiverId()).thenReturn(receiverId1);
+    when(receiver2.getReceiverId()).thenReturn(receiverId2);
+    Map<String, ServiceBusReceiverFacade> receivers =
+        Map.of(
+            receiverId1, receiver1,
+            receiverId2, receiver2
+        );
+
+    //when
+    testObj = new TaskToReceiverBridge(blockingQueue, offsetReader, receivers, executorService);
+    testObj.closeReceivers();
+
+    //then
+    verify(receiver1).unsubscribeAndClose();
+    verify(receiver2).unsubscribeAndClose();
+  }
+
+  @Test
+  void pollShouldDrainAllMessagesFromQueue() throws InterruptedException {
+    //given
+    int arrayBlockingQueueCapacity = 10;
+    String receiverId1 = "RECEIVER1";
+    String messageIdTemplate = "MSGID%s";
+    ExecutorService executorService = mock(ExecutorService.class);
+    ServiceBusReceiverFacade receiver1 = mock(ServiceBusReceiverFacade.class);
+    List<SourceRecord> allSourceRecords = new ArrayList<>(arrayBlockingQueueCapacity);
+    BlockingQueue<ServiceBusMessageHolder> sourceRecordBlockingQueue =
+        new ArrayBlockingQueue<>(arrayBlockingQueueCapacity);
+    Map<String, ServiceBusReceiverFacade> receivers = Map.of(receiverId1, receiver1);
+
+    for (int i = 0; i < arrayBlockingQueueCapacity; i++) {
+      ServiceBusMessageHolder mockedRecord = mock(ServiceBusMessageHolder.class);
+      ServiceBusReceivedMessage busReceivedMessage = mock(ServiceBusReceivedMessage.class);
+      SourceRecord sourceRecord = mock(SourceRecord.class);
+      when(mockedRecord.getTranslatedRecord()).thenReturn(sourceRecord);
+      when(mockedRecord.getOriginalRecord()).thenReturn(busReceivedMessage);
+      when(busReceivedMessage.getMessageId()).thenReturn(String.format(messageIdTemplate, i));
+      allSourceRecords.add(sourceRecord);
+      sourceRecordBlockingQueue.put(mockedRecord);
+    }
+
+    //when
+    testObj = new TaskToReceiverBridge(sourceRecordBlockingQueue, offsetReader, receivers, executorService);
+    List<SourceRecord> polled = testObj.poll();
+
+    //then
+    assertThat(polled).hasSize(arrayBlockingQueueCapacity);
+    assertThat(polled).containsExactlyElementsOf(allSourceRecords);
```

Comment:
This should really be joined together in a single assertion chain:

`    assertThat(polled).hasSize(arrayBlockingQueueCapacity).containsExactlyElementsOf(allSourceRecords);
`

Reply:
okay